oak_runtime/src/node/http/client.rs

```diff
@@ -0,0 +1,327 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! HTTP client pseudo-Node that allows sending HTTP requests to a specific external service.
+
+use crate::{
+    io::{Receiver, ReceiverExt},
+    node::{invocation::InvocationExt, Node},
+    proto::oak::invocation::HttpInvocation as Invocation,
+    NodePrivilege, RuntimeProxy,
+};
+use hyper::body::to_bytes;
+use log::{debug, error, info, warn};
+use maplit::hashset;
+use oak_abi::{
+    proto::oak::application::{HttpClientConfiguration, UriScheme},
+    Handle, OakStatus,
+};
+use oak_io::{handle::ReadHandle, OakError};
+use oak_services::proto::oak::encap::{HeaderMap, HttpResponse};
+use std::io;
+use tokio::sync::oneshot;
+
+type HyperClient =
+    hyper::client::Client<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>, hyper::Body>;
+
+/// Struct that represents an HTTP client pseudo-Node.
+pub struct HttpClientNode {
+    /// Pseudo-Node name.
+    node_name: String,
+    /// HTTP client to allow re-use of connection across multiple method invocations. It can be
+    /// used to send either HTTP or HTTPS requests.
+    http_client: HyperClient,
+    /// The authority corresponding to this client, or none in case of a public client.
+    authority: Option<String>,
```

Comment:
Does this also mean that this is None if the scheme was HTTP, or the value of the authority in the protobuf if it was HTTPS? That seems somewhat inconsistent, for instance below I would expect to see a check on both scheme and authority, but only authority is checked, which I guess is because the scheme has been already taken into account?